core/domain/state_domain.py

```diff
@@ -1925,6 +1928,341 @@ def validate(self) -> None:
                 self.duration_secs)
 
 
+def validate_rte_tags(
+    html: str, is_tags_nested_inside_tabs_or_collapsible: bool = False
+) -> None:
+    """Validate all the RTE tags.
+
+    Args:
+        html: str. The RTE content of the state.
+        is_tags_nested_inside_tabs_or_collapsible: bool. True when we
+            validate tags inside `Tabs` or `Collapsible` tag.
+
+    Raises:
+        utils.ValidationError. Raises the validation error in case the
+            content is not valid.
+    """
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+
+        if len(tag['alt-with-value'].strip()) < 5:
+            raise utils.ValidationError(
+                'Image tag \'alt-with-value\' attribute should not '
+                'be less than 5.'
+            )
+
+        if len(tag['content-with-value'].strip()) > 500:
+            raise utils.ValidationError(
+                'Image tag \'content-with-value\' attribute should not '
+                'be greater than 500 characters.'
+            )
+
+        if not tag.has_attr('filepath-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'filepath-with-value\' attribute.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-skillreview'):
+
+        if not tag.has_attr('text-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'text-with-value\' '
+                'attribute.'
+            )
+
+        if tag['text-with-value'] is None:
+            raise utils.ValidationError(
+                'SkillReview tag should not have \'text-with-value\' '
+                'attribute value as None.'
+            )
+
+        if (
+            tag['text-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
```

Comment:
you use these 4 "empty strings" in a few places. Can you just declare it as a variable once and then reuse it

Reply:
Done